sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServiceSyncContext.cs

```diff
@@ -303,6 +303,28 @@ public Task CancelAndUpdateItemAsync(MobileServiceTableOperationError error, JOb
             });
         }
 
+        public Task UpdateOperationAsync(MobileServiceTableOperationError error, JObject item)
+        {
+            string itemId = error.Item.Value<string>(MobileServiceSystemColumns.Id);
```

Comment:
Checked all the code paths, error.item is populated from operation.I followed code pattern for cancelAndUpdate() and cancelAndDiscard() and also ran through all the scenarios for conflicts to verify that it is always populated.